w3-blob.md

```diff
@@ -0,0 +1,199 @@
+| Agent       | A [Principal] identified by [`did:key`] identifier, representing a user in an application. |
+| Issuer | A [principal] delegating capabilities to another [principal]. It is the signer of the [UCAN]. Specified in the `iss` field of a UCAN. |
+| Audience | Principal access is shared with. Specified in the `aud` field of a UCAN. |
+
+
+### Space
+
+A namespace, often referred as a "space", is an owned resource that can be shared. It corresponds to a unique asymmetric cryptographic keypair and is identified by a [`did:key`] URI.
+
+### Blob
+
+Blob is a fixed size byte array addressed by the [multihash]. Usually blobs are used to represent set of IPLD blocks at different byte ranges.
+
+# Capabilities
+
+## Add Blob
+
+Authorized agent MAY invoke `/space/content/add/blob` capability on the [space] subject to store specific byte array.
+
+> Note that storing a blob does not imply advertising it on the network or making it publicly available.
+
+### Add Blob Capability
+
+#### Add Blob Capability Schema
+
+```ipldsch
+type struct AddBlob {
+  cmd   "/space/content/add/blob"
+  sub   SpaceDID
+  args  Blob
+}
+
+type struct Blob {
+  content   Multihash
+  size      Int
+}
+```
+
+#### Blob Content
+
+Blob `content` field MUST be a [multihash] digest of the blob payload bytes, uniquely identifying blob.
+
+#### Blob Size
+
+Blob `size` field MUST be set to the size of the blob in bytes.
+
+### Add Blob Receipt
+
+#### Add Blob Result
+
+```ipldsch
+type BlobAddResult union {
+  BlobAddSuccess    "ok"
+  BlobAddFailure    "error"
+} representation keyed
+
+type BlobAddSuccess union {
+  BlobAddAllocation     "allocated"
+  &UCANLocationClaim    "committed"
+} representation keyed
+
+type BlobAddAllocation struct {
+  content   Multihash
+  size      Int
+
+  url       URL
+  headers   HTTPHeaders
+}
+
+type UCANLocationClaim struct {
+  iss     DID
+  aud     DID
+  sub     SpaceDID
+  cmd     "assert/location"
+  args    LocationClaim
+  nonce   bytes
+  exp     Int
+}
+
+type LocationClaim struct {
+  content    Multihash
+  url        URL
+  range      ByteRange
+}
+
+type ByteRange struct {
+  offset    Int
+  length    Int
+} representation tuple
+```
+
+#### Add Blob Success
+
+##### Add Blob Allocated
+
+Capability provider MUST issue receipt with `BlobAddAllocation` success result if underlying space has enough capacity for the requested blob, but has not such blob locally.
+
+The `url` field MUST be set to the HTTP PUT endpoint where `content` matching specified [multihash] and `size` could be uploaded.
+
+The `headers` field MUST be set to the HTTP headers to be send along with `content` to the HTTP PUT endpoint.
+
+###### Add Blob Committed
+
+Capability provider MUST issue receipt with `LocationClaim` capability delegated to the [space] when requested blob is added to the space, which can happen if provider already has a matching blob locally.
```

Comment:
I am confused here. This should only happen when bytes make its way to the bucket right? (based also on the diagram you previously shared)

Reply:
I think this is poorly worded, I'll fix it but what I meant to say that if provider already has such blob they can issue `"committed"` case of the ok result as opposed to `"allocated"`.

You are right that if provider does not already have a blob we are not going to issue `LocationClaim` until we know we have it.